Code/GraphMol/Depictor/RDDepictor.cpp

```diff
@@ -451,57 +450,160 @@ unsigned int compute2DCoordsMimicDistMat(
+  double d_sumIndices;
+};
+
+void generateDepictionMatching2DStructure(RDKit::ROMol &mol,
+                                          const RDKit::ROMol &reference,
+                                          const RDKit::MatchVectType &matchVect,
+                                          const std::vector<int> &refMatch,
+                                          int confId, bool forceRDKit) {
+  RDGeom::INT_POINT2D_MAP coordMap;
+  const RDKit::Conformer &conf = reference.getConformer(confId);
+  for (const auto &mv : matchVect) {
+    if (refMatch[mv.first] < 0) {
+      throw RDDepict::DepictException("Found unexpected -1 in refMatch vector");
+    }
+    RDGeom::Point3D pt3 = conf.getAtomPos(refMatch[mv.first]);
+    RDGeom::Point2D pt2(pt3.x, pt3.y);
+    coordMap[mv.second] = pt2;
+  }
+  RDDepict::compute2DCoords(mol, &coordMap, false /* canonOrient */,
+                            true /* clearConfs */, 0, 0, 0, false, forceRDKit);
+}
+}  // namespace detail
+
+//! \brief Compute 2D coordinates where a piece of the molecule is
+//   constrained to have the same coordinates as a reference;
+//   correspondences between reference and molecule atom indices
+//   are determined by refMatchVect
+void generateDepictionMatching2DStructure(
+    RDKit::ROMol &mol, const RDKit::ROMol &reference,
+    const RDKit::MatchVectType &refMatchVect, int confId, bool forceRDKit) {
+  std::vector<int> refMatch(reference.getNumAtoms(), -1);
+  if (refMatchVect.size() > reference.getNumAtoms(true)) {
+    throw RDDepict::DepictException(
+        "When a refMatchVect is provided, it must have size "
+        "<= number of atoms in the reference");
+  }
+  for (auto &i : refMatchVect) {
+    if (i.first >= refMatch.size()) {
+      throw RDDepict::DepictException("Index in refMatchVect out of range");
+    }
+    refMatch[i.first] = i.first;
+  }
+  detail::generateDepictionMatching2DStructure(mol, reference, refMatchVect,
+                                               refMatch, confId, forceRDKit);
+}
+
 //! \brief Compute 2D coordinates where a piece of the molecule is
 //   constrained to have the same coordinates as a reference.
 void generateDepictionMatching2DStructure(RDKit::ROMol &mol,
                                           const RDKit::ROMol &reference,
                                           int confId,
                                           RDKit::ROMol *referencePattern,
-                                          bool acceptFailure, bool forceRDKit) {
+                                          bool acceptFailure, bool forceRDKit,
+                                          bool allowRGroups) {
   std::vector<int> refMatch;
   RDKit::MatchVectType matchVect;
+  RDKit::MatchVectType matchVectFiltered;
+  auto &matchVectRef = matchVect;
   if (referencePattern) {
-    if (reference.getNumAtoms(true) != referencePattern->getNumAtoms(true)) {
+    if (referencePattern->getNumAtoms(true) > reference.getNumAtoms(true)) {
```

Comment:
`getNumAtoms(true)` is the same as `getNumAtoms()`, might as well go ahead and use the simpler (and more commonly used) form.

Reply:
Yeah, that was just a copy/paste from previous code. Done.